Python/compile.c

```diff
@@ -8097,52 +8118,59 @@ optimize_basic_block(struct compiler *c, basicblock *bb, PyObject *consts)
         else {
             target = &nop;
         }
-        switch (inst->i_opcode) {
+        if (HAS_CONST(inst->i_opcode)) {
```

Comment:
Why move this out of the switch?
This code needs to understand the opcodes anyway, so adding multiple cases is fine.
```
    case LOAD_CONST:
    case LOAD_CONST_COMMON:
    case LOAD_INT:
        ...
```

Reply:
We could leave it in the switch. Though it will be less obvious if you forget to add a new opcode to this case.

Reply:
Actually it's not true that this code needs to understand the individual load-const opcodes - this is now abstracted away in get_const_value().

Reply:
It still needs to understand all the other instructions. Handling all of them is more regular than handling almost all of them.

I prefer a switch statement here as each case is clearly labelled and I don't need to lookup what `HAS_CONST` does.